lib/topics_filter.rb

```diff
@@ -81,19 +81,29 @@ def filter_status(status:, category_id: nil)
   private
 
   def filter_categories(category_slugs:, exclude_subcategories: false)
+    valid_category_ids = []
+    valid_category_slugs = []
+
     category_ids =
-      Category
-        .where(slug: category_slugs)
-        .filter { |category| @guardian.can_see_category?(category) }
-        .map(&:id)
+      category_slugs.flat_map { |category_slug| Category.ids_from_slugs(category_slug.split(":")) }
+
+    Category
+      .where(id: category_ids)
+      .filter { |category| @guardian.can_see_category?(category) }
+      .each do |category|
+        valid_category_ids << category.id
+        valid_category_slugs << category.slug
+      end
 
-    return @scope.none if category_ids.length != category_slugs.length
+    # Don't return any records if the user does not have access to any of the categories
+    return @scope.none if valid_category_slugs.length < category_slugs.length
```

Comment:
Technically, we don't need `valid_category_slugs` as we just need length of valid . Maybe let's use `valid_category_ids.length` ?

Reply:
Good catch :+1: Have removed the unneeded variable.